### DISCUSSION TITLE
Use token server side?

### DISCUSSION THREAD

**Discussion by realJogicodes** [pocketbase/pocketbase#4836] [Q&A]
Hi

can I use the auth token server side to authenticate users? Say I login client side, store the token in a cookie like so:
```
// login/+page.svelte
const authData = await pb.collection('users').authWithPassword(email, passwd);
document.cookie = `pocketbase_auth_token=${authData.token}; path=/; expires=${69420}`;
goto('/next/route');
```

I've got it now in the load function:

```
export function load({ cookies }) {
	const authToken = cookies.get('pocketbase_auth_token');
	console.log(authToken);
	if (!authToken) {
		throw { status: 403 };
	}
       // validate token
}
```
Can I validate the token against pocketbase users somehow server side?
So far, I am doing client side redirects, I am curious if this can be offloaded upwards to the server for more control

**Comment by ganigeorgiev** [comment]
You can load it into the store with `pb.authStore.save(token)`.
To validate the token you can send ["Auth refresh" call](https://pocketbase.io/docs/api-records/#auth-refresh).

```js
pb.authStore.save(authToken)

try {
    // note that "isValid" only loosely checks whether the store token is valid, aka. it is a JWT with non-expired "exp" claim, but without verifying the signature
    pb.authStore.isValid && await pb.collection("users").authRefresh();
} catch {
    // invalid or expired token
    pb.authStore.clear();
}
```

**Reply by ganigeorgiev** [reply]
_Side-note: I recommend specifying a `SameSite` attribute to your cookie. An alternative could be to use the JS SDK `exportToCookie` helper:_
```js
// by default exported as "pb_auth" cookie and Path=/, Expires=..., SameSite=Strict, Secure=true and HttpOnly=true attributes
document.cookie = pb.authStore.exportToCookie({ httpOnly: false }, "your_custom_cookie_name")
```

**Reply by realJogicodes** [reply]
Cool, thanks for getting back so quickly. Curious if I've got this right, with your suggested implementation I put the server response into the try{} block and if there's no allowed user, anything catch{} executes so if I send the data from within try, it's locked down and in catch {} I inform users and maybe redirect to signup or login?

**Reply by ganigeorgiev** [reply]
I don't understand your flow and the reason why you want to use a JS server response in the first place. I'm getting tired by this JS SSR related questions and unless you have a very specific reason I really don't recommend using PocketBase that way as most likely you'll mess up the security. You can find some ramblings on this in https://github.com/pocketbase/pocketbase/discussions/3792#discussioncomment-7620032.

**Reply by realJogicodes** [reply]
That's good to know. A am coming from flask which handlespermissions on the server side using the @login_required decorator, I guess I want to shoe horn pb into doing the same thing out of habit, mea culpa. Side note, it does not help your case that there are a couple youtube tutorials that do exactly that, server side auth, maybe leave a comment under them to discourage the use of that and guilt the youtube influenzoor into releasing an updated version. 

onMount redirect if authstore invalid is good enough if the API permissions are set up properly I guess.
